### PR TITLE
Create unit test for function bitmap_name

### PR DESCRIPTION
**Pull request by sam-stajnko**
# Description

This PR includes unit tests for the Splashkit function bitmap_name. Tests include making sure that the returned name matches the name of the bitmap and checking if bitmap_name returns an empty string when an invalid bitmap reference is passed as a parameter.

## Type of change

- [x] New unit test added, with multiple sections

## How Has This Been Tested?

To test, build the project and run the skunit_tests executable. Steps are provided on the Splashkit Documentation website if you are uncertain.

The test is named "name can be retrieved from bitmap" and should pass. If the test fails, let me know and we can track down the problem together.

## Testing Checklist

- [x] Tested with skunit_tests

Screenshot from 2025-08-28 16-03-57

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review from ... on the Pull Request

### CODE REVIEW

File: coresdk/src/test/unit_tests/unit_test_bitmap.cpp [hunk 1]
```diff
@@ -128,3 +128,62 @@ TEST_CASE("bitmap bounding details can be retrieved", "[bitmap]")
     }
     free_bitmap(bmp);
 }
+
+TEST_CASE("name can be retrieved from bitmap", "[bitmap][bitmap_name]")
+{   
+    SECTION("can get bitmap name")
+    {
+        bitmap bmp = load_bitmap("rocket", "rocket_sprt.png");
```
Comment (dijidiji) [inline coresdk/src/test/unit_tests/unit_test_bitmap.cpp]: Since there's already a test for loading a bitmap from a file, it's not necessary to load a bitmap here and we could just create a new bitmap with an arbitrary name.

I think it's best practice to try and isolate the functionality/workflow we're testing as much as possible, so in this case we can avoid touching the file system at all.

File: coresdk/src/test/unit_tests/unit_test_bitmap.cpp [hunk 2]
```diff
@@ -128,3 +128,62 @@ TEST_CASE("bitmap bounding details can be retrieved", "[bitmap]")
...
+
+    SECTION("empty string returned for bitmap name when passing nullptr")
+    {
+        string name = bitmap_name(nullptr);
+        REQUIRE(name == "");
+    }
+
+    SECTION("empty string returned for bitmap when passing freed bitmap")
+    {
+        bitmap bmp = load_bitmap("frog", "frog.png");
```
Comment (dijidiji) [inline coresdk/src/test/unit_tests/unit_test_bitmap.cpp]: Similarly, we could create a blank sprite and free it, since a previous test case covers the freeing of loaded bitmaps.

File: coresdk/src/test/unit_tests/unit_test_bitmap.cpp [hunk 3]
```diff
@@ -128,3 +128,62 @@ TEST_CASE("bitmap bounding details can be retrieved", "[bitmap]")
...
+        REQUIRE(name == "");
+    }
+
+    SECTION("can get name of bitmap when name includes special characters")
+    {
+        string silly_name = "AbCdEf 2+3=5 !@#$% \n";
+        bitmap bmp = create_bitmap(silly_name, 128, 128);
+
+        string returned_name = bitmap_name(bmp);
+        REQUIRE(returned_name == silly_name);
```
Comment (dijidiji) [inline coresdk/src/test/unit_tests/unit_test_bitmap.cpp:196]: You can also use [Catch2's string matchers](https://github.com/catchorg/Catch2/blob/v2.x/docs/matchers.md) for assertions with std::strings.

For example, to use Equals:
```cpp
using Catch::Matchers::Equals;
```

```suggestion
        REQUIRE_THAT(returned_name, Equals(silly_name));
```
I'm not sure if there's a reason to prefer using one over the other, just thought it was worth noting. So don't consider this a required change - I'm half reminding myself here for things to note down in the guide!

**Review by dijidiji — CHANGES_REQUESTED** [review]
The tests work for me, even in isolation and when run in a random order. I think you've considered a lot of edge cases here, which is great.

I could see a case made for keeping the file loading in if you deem it to be important to the test in question.

**Comment by sam-stajnko** [comment]
Hi @dijidiji ,

I have replaced load_bitmap with create_bitmap so there shouldn't be any more issues. As for your suggestion to use REQUIRE_THAT instead of REQUIRE, I have left the code unchanged because I think REQUIRE looks a bit cleaner.

Note, I have rebuilt skunit_tests and ran my tests for bitmap_name and all tests pass, indicating that it is working correctly.
Screenshot from 2025-09-02 12-09-36
If you have anything else that you think needs changing, don't hesitate to ask.
REgards,

Sam Stajnko.

**Review by dijidiji — APPROVED** [review]
Looks good, tests pass on my end as well! Approved.

**Review by Broccoli811 — APPROVED** [review]
The new TEST_CASE for the bitmap_name function looks great. It clearly covers relevant scenarios including valid bitmaps, nullptr, freed bitmaps, dupe names, empty strings, and names with special characters.

I have run a test in my local machine, and the expected results matched the wanted behavior.

For my suggestion, perhaps consider extending the naming logic (blank, blank0, blank1, blank2 etc.) so we can see if it scales predictably, you can also extend this to freed bitmaps where we free multiple at once to confirm if multiple maps are updated correctly. Other than that, I can't think of any other scenarios.

Overall, very good implementation! Great Job! :)
